From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Android Components - Disable crash reporting

Disables and prevents the collection/submission of crash reports.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
index 2b16913688f6e..c115de2c586a1 100644
--- a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
+++ b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/CrashReporter.kt
@@ -32,10 +32,10 @@ import mozilla.components.lib.crash.db.toReportEntity
 import mozilla.components.lib.crash.handler.ExceptionHandler
 import mozilla.components.lib.crash.notification.CrashNotification
 import mozilla.components.lib.crash.prompt.CrashPrompt
-import mozilla.components.lib.crash.service.CrashReporterService
+/* import mozilla.components.lib.crash.service.CrashReporterService
 import mozilla.components.lib.crash.service.CrashTelemetryService
 import mozilla.components.lib.crash.service.SendCrashReportService
-import mozilla.components.lib.crash.service.SendCrashTelemetryService
+import mozilla.components.lib.crash.service.SendCrashTelemetryService */
 import mozilla.components.support.base.log.logger.Logger
 
 /**
@@ -99,8 +99,8 @@ private const val START_OF_144_NIGHTLY_TIMESTAMP = 1755475200000L
  * @param useLegacyReporting Enable/Disable handling crash reporting through a notification or system dialog.
  */
 class CrashReporter internal constructor(
-    private val services: List<CrashReporterService> = emptyList(),
-    private val telemetryServices: List<CrashTelemetryService> = emptyList(),
+    private val services: List<Any> = emptyList(),
+    private val telemetryServices: List<Any> = emptyList(),
     private val shouldPrompt: Prompt = Prompt.NEVER,
     enabled: Boolean = true,
     internal val promptConfiguration: PromptConfiguration = PromptConfiguration(),
@@ -115,8 +115,8 @@ class CrashReporter internal constructor(
 
     constructor(
         context: Context,
-        services: List<CrashReporterService> = emptyList(),
-        telemetryServices: List<CrashTelemetryService> = emptyList(),
+        services: List<Any> = emptyList(),
+        telemetryServices: List<Any> = emptyList(),
         shouldPrompt: Prompt = Prompt.NEVER,
         enabled: Boolean = true,
         promptConfiguration: PromptConfiguration = PromptConfiguration(),
@@ -154,9 +154,9 @@ class CrashReporter internal constructor(
         get() = runtimeTagProviders.fold(emptyMap()) { acc, provider -> acc + provider() }
 
     init {
-        require(services.isNotEmpty() || telemetryServices.isNotEmpty()) {
+/*        require(services.isNotEmpty() || telemetryServices.isNotEmpty()) {
             "No crash reporter services defined"
-        }
+        } */
     }
 
     /**
@@ -165,8 +165,8 @@ class CrashReporter internal constructor(
     fun install(
         applicationContext: Context,
         handleCaughtExceptionSideEffects: (() -> Unit)? = null,
-    ): CrashReporter {
-        instance = this
+    ) {
+/*        instance = this
 
         val defaultHandler = Thread.getDefaultUncaughtExceptionHandler()
         val handler = ExceptionHandler(
@@ -177,7 +177,7 @@ class CrashReporter internal constructor(
         )
         Thread.setDefaultUncaughtExceptionHandler(handler)
 
-        return this
+        return this */
     }
 
     /**
@@ -224,7 +224,7 @@ class CrashReporter internal constructor(
      */
     fun submitReport(crash: Crash, then: () -> Unit = {}): Job {
         return scope.launch {
-            services.forEach { service ->
+/*            services.forEach { service ->
                 val reportId = when (crash) {
                     is Crash.NativeCodeCrash -> service.report(crash)
                     is Crash.UncaughtExceptionCrash -> service.report(crash)
@@ -242,7 +242,7 @@ class CrashReporter internal constructor(
             logger.info("Crash report submitted to ${services.size} services")
             withContext(mainDispatcher) {
                 then()
-            }
+            } */
         }
     }
 
@@ -251,7 +251,7 @@ class CrashReporter internal constructor(
      */
     fun submitCrashTelemetry(crash: Crash, then: () -> Unit = {}): Job {
         return scope.launch {
-            telemetryServices.forEach { telemetryService ->
+/*            telemetryServices.forEach { telemetryService ->
                 when (crash) {
                     is Crash.NativeCodeCrash -> telemetryService.record(crash)
                     is Crash.UncaughtExceptionCrash -> telemetryService.record(crash)
@@ -261,7 +261,7 @@ class CrashReporter internal constructor(
             logger.info("Crash report submitted to ${telemetryServices.size} telemetry services")
             withContext(mainDispatcher) {
                 then()
-            }
+            } */
         }
     }
 
@@ -273,16 +273,16 @@ class CrashReporter internal constructor(
          * if stacktrace is empty, replace throwable with UnexpectedlyMissingStacktrace exception so
          * we can figure out which module is submitting caught exception reports without a stacktrace.
          */
-        var reportThrowable = throwable
+/*        var reportThrowable = throwable
         if (throwable.stackTrace.isEmpty()) {
             reportThrowable = CrashReporterException.UnexpectedlyMissingStacktrace("Missing Stacktrace", throwable)
         }
 
-        logger.info("Caught Exception report submitted to ${services.size} services")
+        logger.info("Caught Exception report submitted to ${services.size} services") */
         return scope.launch {
-            services.forEach {
+/*            services.forEach {
                 it.report(reportThrowable, crashBreadcrumbsCopy())
-            }
+            } */
         }
     }
 
@@ -296,14 +296,14 @@ class CrashReporter internal constructor(
      * ```
      */
     override fun recordCrashBreadcrumb(breadcrumb: Breadcrumb) {
-        crashBreadcrumbs.add(breadcrumb)
+//        crashBreadcrumbs.add(breadcrumb)
     }
 
     /**
      * Set whether telemetry has been enabled.
      */
     fun setTelemetryEnabled(enabled: Boolean) {
-        telemetryServices.forEach { it.setTelemetryEnabled(enabled) }
+//        telemetryServices.forEach { it.setTelemetryEnabled(enabled) }
     }
 
     /**
@@ -397,13 +397,13 @@ class CrashReporter internal constructor(
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun sendCrashReport(context: Context, crash: Crash) = try {
-        ContextCompat.startForegroundService(context, SendCrashReportService.createReportIntent(context, crash))
+    internal fun sendCrashReport(context: Context, crash: Crash) {
+/*        ContextCompat.startForegroundService(context, SendCrashReportService.createReportIntent(context, crash))
     } catch (e: BadParcelableException) {
-        (crash as? Crash.UncaughtExceptionCrash)?.let {
+        (crash as? Crash.UncaughtExceptionCrash)?.let { */
             // We may end up with a throwable that isn't completely serializable, which will cause
             // a crash when the service tries to unbundle it.
-            val updatedCrash = it.copy(throwable = it.throwable.forceSerializable())
+/*            val updatedCrash = it.copy(throwable = it.throwable.forceSerializable())
             ContextCompat.startForegroundService(
                 context,
                 SendCrashReportService.createReportIntent(context, updatedCrash),
@@ -417,17 +417,17 @@ class CrashReporter internal constructor(
             logger.warn("ignored failed service start while backgrounded")
         } else {
             throw e
-        }
+        } */
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun sendCrashTelemetry(context: Context, crash: Crash) = try {
-        ContextCompat.startForegroundService(context, SendCrashTelemetryService.createReportIntent(context, crash))
+    internal fun sendCrashTelemetry(context: Context, crash: Crash) {
+/*        ContextCompat.startForegroundService(context, SendCrashTelemetryService.createReportIntent(context, crash))
     } catch (e: BadParcelableException) {
-        (crash as? Crash.UncaughtExceptionCrash)?.let {
+        (crash as? Crash.UncaughtExceptionCrash)?.let { */
             // We may end up with a throwable that isn't completely serializable, which will cause
             // a crash when the service tries to unbundle it.
-            val updatedCrash = it.copy(throwable = it.throwable.forceSerializable())
+/*            val updatedCrash = it.copy(throwable = it.throwable.forceSerializable())
             ContextCompat.startForegroundService(
                 context,
                 SendCrashTelemetryService.createReportIntent(context, updatedCrash),
@@ -441,7 +441,7 @@ class CrashReporter internal constructor(
             logger.warn("ignored failed service start while backgrounded")
         } else {
             throw e
-        }
+        } */
     }
 
     @VisibleForTesting
@@ -464,8 +464,8 @@ class CrashReporter internal constructor(
         }
     }
 
-    internal fun getCrashReporterServiceById(id: String): CrashReporterService? {
-        return services.firstOrNull { it.id == id }
+    internal fun getCrashReporterServiceById(id: String) {
+//        return services.firstOrNull { it.id == id }
     }
 
     enum class Prompt {
diff --git a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/ui/DisplayableCrash.kt b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/ui/DisplayableCrash.kt
index 8c4943f23067..7ef2293767a7 100644
--- a/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/ui/DisplayableCrash.kt
+++ b/mobile/android/android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/ui/DisplayableCrash.kt
@@ -60,7 +60,7 @@ internal fun CrashWithReports.toCrash(reporter: CrashReporter) = DisplayableCras
 
 internal fun ReportEntity.toReport(reporter: CrashReporter): DisplayableCrash.Report {
     val service = reporter.getCrashReporterServiceById(serviceId)
-    val name = service?.name ?: serviceId
-    val url = service?.createCrashReportUrl(reportId)
+    val name = ""
+    val url = ""
     return DisplayableCrash.Report(name, url)
 }
diff --git a/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt b/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
index ae8c5002251e..fa7e3d50695a 100644
--- a/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
+++ b/mobile/android/android-components/components/support/appservices/src/main/java/mozilla/components/support/AppServicesInitializer.kt
@@ -5,7 +5,7 @@
 package mozilla.components.support
 
 import mozilla.appservices.RustComponentsInitializer
-import mozilla.components.concept.base.crash.CrashReporting
+// import mozilla.components.concept.base.crash.CrashReporting
 import mozilla.components.support.base.log.Log
 import mozilla.components.support.rustlog.RustLog
 
@@ -41,7 +41,7 @@ object AppServicesInitializer {
      */
     data class Config(
         // Do not add a crash reporting default. We want to be explicit if we make this choice.
-        val crashReporting: CrashReporting?,
+//        val crashReporting: CrashReporting?,
         val logLevel: Log.Priority = Log.Priority.INFO,
     )
 }
-- 
