From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Android Components - Display built-in add-ons

Displays built-in add-ons alongside user-installed ones.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt b/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt
index 0575b1166697..6470f6a7f64d 100644
--- a/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt
+++ b/mobile/android/android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt
@@ -127,7 +127,7 @@ class AddonManager(
 
             // Build a list of installed extensions that are not built-in extensions.
             val installedAddons = installedExtensions
-                .filterValues { !it.isBuiltIn() }
+//                .filterValues { !it.isBuiltIn() }
                 .map {
                     val extension = it.value
                     val installedState = toInstalledState(extension)
-- 
