From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Android Components - Remove Nimbus

Removes the Nimbus library/service.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/android-components/components/browser/engine-gecko/build.gradle b/mobile/android/android-components/components/browser/engine-gecko/build.gradle
index 8b57d01c4fc8..87e90956d26f 100644
--- a/mobile/android/android-components/components/browser/engine-gecko/build.gradle
+++ b/mobile/android/android-components/components/browser/engine-gecko/build.gradle
@@ -21,7 +21,7 @@ buildscript {
 
 plugins {
     alias(libs.plugins.python.envs.plugin)
-    id 'org.mozilla.nimbus-gradle-plugin'
+//    id 'org.mozilla.nimbus-gradle-plugin'
 }
 
 apply plugin: 'com.android.library'
@@ -59,9 +59,9 @@ dependencies {
 
     implementation project(':components:concept-engine')
     implementation project(':components:concept-fetch')
-    implementation(project(':components:service-nimbus')) {
+/*    implementation(project(':components:service-nimbus')) {
         exclude group: 'org.mozilla.telemetry', module: 'glean-native'
-    }
+    } */
     implementation project(':components:support-ktx')
     implementation project(':components:support-utils')
 
@@ -96,7 +96,7 @@ ext {
 apply plugin: "org.mozilla.telemetry.glean-gradle-plugin"
 apply from: '../../../common-config.gradle'
 apply from: '../../../publish.gradle'
-nimbus {
+/* nimbus {
     // The path to the Nimbus feature manifest file
     manifestFile = "geckoview.fml.yaml"
 
@@ -110,7 +110,7 @@ nimbus {
     // *NOTE*: This example will not work for all projects, but should work for Fenix, Focus, and Android Components
     applicationServicesDir = gradle.hasProperty('localProperties.autoPublish.application-services.dir')
             ? gradle.getProperty('localProperties.autoPublish.application-services.dir') : null
-}
+} */
 ext.configurePublish(config.componentsGroupId, project.name, project.ext.description)
 
 // Non-official versions are like "61.0a1", where "a1" is the milestone.
-- 
