From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Application Services - Remove Glean

Removes the Glean SDK.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt b/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt
index 5a7aaab55..2b900c888 100644
--- a/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt
+++ b/components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt
@@ -26,11 +26,11 @@ import kotlinx.coroutines.async
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 import mozilla.appservices.remotesettings.RemoteSettingsService
-import mozilla.telemetry.glean.Glean
+// // import mozilla.telemetry.glean.Glean
 import org.json.JSONObject
-import org.mozilla.experiments.nimbus.GleanMetrics.NimbusEvents
-import org.mozilla.experiments.nimbus.GleanMetrics.NimbusHealth
-import org.mozilla.experiments.nimbus.GleanMetrics.Pings
+// /* import org.mozilla.experiments.nimbus.GleanMetrics.NimbusEvents
+// import org.mozilla.experiments.nimbus.GleanMetrics.NimbusHealth
+// import org.mozilla.experiments.nimbus.GleanMetrics.Pings */
 import org.mozilla.experiments.nimbus.internal.AppContext
 import org.mozilla.experiments.nimbus.internal.AvailableExperiment
 import org.mozilla.experiments.nimbus.internal.DatabaseLoadExtraDef
@@ -95,7 +95,7 @@ open class Nimbus(
 
     private val metricsHandler = object : MetricsHandler {
         override fun recordDatabaseLoad(event: DatabaseLoadExtraDef) {
-            NimbusEvents.databaseLoad.record(
+/*            NimbusEvents.databaseLoad.record(
                 NimbusEvents.DatabaseLoadExtra(
                     corrupt = event.corrupt,
                     initialVersion = event.initialVersion?.toInt(),
@@ -103,20 +103,20 @@ open class Nimbus(
                     migratedVersion = event.migratedVersion?.toInt(),
                     migrationError = event.migrationError,
                 ),
-            )
+            ) */
         }
         override fun recordDatabaseMigration(event: DatabaseMigrationExtraDef) {
-            NimbusEvents.databaseMigration.record(
+/*            NimbusEvents.databaseMigration.record(
                 NimbusEvents.DatabaseMigrationExtra(
                     reason = event.reason,
                     fromVersion = event.fromVersion.toInt(),
                     toVersion = event.toVersion.toInt(),
                     error = event.error,
                 ),
-            )
+            ) */
         }
         override fun recordEnrollmentStatuses(enrollmentStatusExtras: List<EnrollmentStatusExtraDef>) {
-            for (extra in enrollmentStatusExtras) {
+/*            for (extra in enrollmentStatusExtras) {
                 NimbusEvents.enrollmentStatus.record(
                     NimbusEvents.EnrollmentStatusExtra(
                         branch = extra.branch,
@@ -127,42 +127,42 @@ open class Nimbus(
                         conflictSlug = extra.conflictSlug,
                     ),
                 )
-            }
+            } */
         }
 
         override fun recordFeatureActivation(event: FeatureExposureExtraDef) {
-            NimbusEvents.activation.record(
+/*            NimbusEvents.activation.record(
                 NimbusEvents.ActivationExtra(
                     experiment = event.slug,
                     branch = event.branch,
                     featureId = event.featureId,
                 ),
-            )
+            ) */
         }
 
         override fun recordFeatureExposure(event: FeatureExposureExtraDef) {
-            NimbusEvents.exposure.record(
+/*            NimbusEvents.exposure.record(
                 NimbusEvents.ExposureExtra(
                     experiment = event.slug,
                     branch = event.branch,
                     featureId = event.featureId,
                 ),
-            )
+            ) */
         }
 
         override fun recordMalformedFeatureConfig(event: MalformedFeatureConfigExtraDef) {
-            NimbusEvents.malformedFeature.record(
+/*            NimbusEvents.malformedFeature.record(
                 NimbusEvents.MalformedFeatureExtra(
                     experiment = event.slug,
                     branch = event.branch,
                     featureId = event.featureId,
                     partId = event.part,
                 ),
-            )
+            ) */
         }
 
         override fun submitTargetingContext() {
-            org.mozilla.experiments.nimbus.GleanMetrics.Pings.nimbusTargetingContext.submit()
+// //            org.mozilla.experiments.nimbus.GleanMetrics.Pings.nimbusTargetingContext.submit()
         }
     }
 
@@ -231,11 +231,11 @@ open class Nimbus(
         try {
             nimbusClient.getFeatureConfigVariables(featureId)?.let { JSONObject(it) }
         } catch (e: NimbusException.DatabaseNotReady) {
-            NimbusHealth.cacheNotReadyForFeature.record(
+/*            NimbusHealth.cacheNotReadyForFeature.record(
                 NimbusHealth.CacheNotReadyForFeatureExtra(
                     featureId = featureId,
                 ),
-            )
+            ) */
             null
         } catch (e: Throwable) {
             reportError("getFeatureConfigVariablesJson", e)
@@ -311,9 +311,9 @@ open class Nimbus(
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal fun fetchExperimentsOnThisThread() = withCatchAll("fetchExperiments") {
         try {
-            NimbusHealth.fetchExperimentsTime.measure {
-                nimbusClient.fetchExperiments()
-            }
+            nimbusClient.fetchExperiments()
+//                nimbusClient.fetchExperiments()
+//            }
             updateObserver {
                 it.onExperimentsFetched()
             }
@@ -349,7 +349,7 @@ open class Nimbus(
                 val time = measureTimeMillis {
                     enrollmentChangeEvents = nimbusClient.applyPendingExperiments()
                 }
-                NimbusHealth.applyPendingExperimentsTime.accumulateSingleSample(time)
+//                NimbusHealth.applyPendingExperimentsTime.accumulateSingleSample(time)
 
                 // SAFETY: events is only null at declaration time and is
                 // immediately assigned a non-null value inside the
@@ -407,13 +407,13 @@ open class Nimbus(
     ) {
         val experiments = nimbusClient.getActiveExperiments()
 
-        if (initial) {
+/*        if (initial) {
             // During initialization we need to report the experiment status of
             // all pre-existing experiments.
             for (experiment in experiments) {
                 Glean.setExperimentActive(experiment.slug, experiment.branchSlug)
             }
-        }
+        } */
 
         if (initial || enrollmentChangeEvents.isNotEmpty()) {
             // During initialization we need to inform the application when we've
@@ -619,19 +619,19 @@ open class Nimbus(
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal fun recordExperimentTelemetry(experiments: List<EnrolledExperiment>) {
         // Call Glean.setExperimentActive() for each active experiment.
-        experiments.forEach { experiment ->
+/*        experiments.forEach { experiment ->
             // For now, we will just record the experiment id and the branch id. Once we can call
             // Glean from Rust, this will move to the nimbus-sdk Rust core.
             Glean.setExperimentActive(
                 experiment.slug,
                 experiment.branchSlug,
             )
-        }
+        } */
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal fun recordExperimentTelemetryEvents(enrollmentChangeEvents: List<EnrollmentChangeEvent>) {
-        enrollmentChangeEvents.forEach { event ->
+/*        enrollmentChangeEvents.forEach { event ->
             when (event.change) {
                 EnrollmentChangeEventType.ENROLLMENT -> {
                     NimbusEvents.enrollment.record(
@@ -693,7 +693,7 @@ open class Nimbus(
                     )
                 }
             }
-        }
+        } */
     }
 
     // The exposure event should be recorded when the expected treatment (or no-treatment, such as
diff --git a/components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt b/components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt
index c9078a52f..87f4e5078 100644
--- a/components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt
+++ b/components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt
@@ -13,9 +13,9 @@ package mozilla.appservices.logins
  * on version updates.
  */
 
-import mozilla.telemetry.glean.private.CounterMetricType
+/* import mozilla.telemetry.glean.private.CounterMetricType
 import mozilla.telemetry.glean.private.LabeledMetricType
-import org.mozilla.appservices.logins.GleanMetrics.LoginsStore as LoginsStoreMetrics
+import org.mozilla.appservices.logins.GleanMetrics.LoginsStore as LoginsStoreMetrics */
 
 /**
  * An artifact of the uniffi conversion - a thin-ish wrapper around a
@@ -124,13 +124,13 @@ class DatabaseLoginsStorage(dbPath: String, keyManager: KeyManager) : AutoClosea
 
     @Throws(LoginsApiException::class)
     fun deleteUndecryptableLoginsAndRecordMetrics() {
-        val result = store.deleteUndecryptableRecordsForRemoteReplacement()
+/*        val result = store.deleteUndecryptableRecordsForRemoteReplacement()
         if (result.localDeleted > 0u) {
             LoginsStoreMetrics.localUndecryptableDeleted.add(result.localDeleted.toInt())
         }
         if (result.mirrorDeleted > 0u) {
             LoginsStoreMetrics.mirrorUndecryptableDeleted.add(result.mirrorDeleted.toInt())
-        }
+        } */
     }
 }
 
@@ -140,12 +140,12 @@ enum class KeyRegenerationEventReason {
 
 fun recordKeyRegenerationEvent(reason: KeyRegenerationEventReason) {
     // Avoid the deprecation warning when calling  `record()` without the optional EventExtras param
-    @Suppress("DEPRECATION")
+/*    @Suppress("DEPRECATION")
     when (reason) {
         KeyRegenerationEventReason.Lost -> LoginsStoreMetrics.keyRegeneratedLost.record()
         KeyRegenerationEventReason.Corrupt -> LoginsStoreMetrics.keyRegeneratedCorrupt.record()
         KeyRegenerationEventReason.Other -> LoginsStoreMetrics.keyRegeneratedOther.record()
-    }
+    } */
 }
 
 /**
@@ -158,8 +158,8 @@ fun recordKeyRegenerationEvent(reason: KeyRegenerationEventReason) {
  * This is a convenience wrapper to measure the two in one shot.
  */
 class LoginsStoreCounterMetrics(
-    val count: CounterMetricType,
-    val errCount: LabeledMetricType<CounterMetricType>,
+//    val count: CounterMetricType,
+//    val errCount: LabeledMetricType<CounterMetricType>,
 ) {
     inline fun <U> measure(callback: () -> U): U {
         return measureIgnoring({ false }, callback)
@@ -170,7 +170,7 @@ class LoginsStoreCounterMetrics(
         shouldIgnore: (Exception) -> Boolean,
         callback: () -> U,
     ): U {
-        count.add()
+//        count.add()
         try {
             return callback()
         } catch (e: Exception) {
@@ -179,19 +179,19 @@ class LoginsStoreCounterMetrics(
             }
             when (e) {
                 is LoginsApiException.NoSuchRecord -> {
-                    errCount["no_such_record"].add()
+                     // no-op
                 }
                 is LoginsApiException.Interrupted -> {
-                    errCount["interrupted"].add()
+                     // no-op
                 }
                 is LoginsApiException.InvalidRecord -> {
-                    errCount["invalid_record"].add()
+                     // no-op
                 }
                 is LoginsApiException -> {
-                    errCount["storage_error"].add()
+                     // no-op
                 }
                 else -> {
-                    errCount["__other__"].add()
+                     // no-op
                 }
             }
             throw e
diff --git a/components/places/build.rs b/components/places/build.rs
index fa4b5578a..6faab0f2e 100644
--- a/components/places/build.rs
+++ b/components/places/build.rs
@@ -2,7 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use glean_build::Builder;
+// use glean_build::Builder;
 
 fn main() {
     uniffi::generate_scaffolding("./src/places.udl").unwrap();
@@ -11,10 +11,10 @@ fn main() {
         .as_ref()
         .map(String::as_str)
     {
-        Builder::default()
+/*        Builder::default()
             .file("metrics.yaml")
             .format("rust_sym")
             .generate()
-            .expect("Error generating Glean Rust bindings");
+            .expect("Error generating Glean Rust bindings"); */
     }
 }
diff --git a/components/places/Cargo.toml b/components/places/Cargo.toml
index 76341ae2d..bb649a4fe 100644
--- a/components/places/Cargo.toml
+++ b/components/places/Cargo.toml
@@ -7,9 +7,9 @@ license = "MPL-2.0"
 exclude = ["/android", "/ios"]
 
 [features]
-default = ["glean-sym"]
+# default = ["glean-sym"]
 # Enable metric recording through glean-sym (Android only)
-glean-sym = ["dep:glean-sym"]
+# glean-sym = ["dep:glean-sym"]
 
 [dependencies]
 sync15 = { path = "../sync15" }
@@ -37,7 +37,7 @@ uniffi = { version = "0.31" }
 
 # glean-sym is only used on Android and iOS.
 [target.'cfg(any(target_os = "android", target_os = "ios"))'.dependencies]
-glean-sym = { git = "https://github.com/mozilla/glean", tag = "v68.0.0", optional = true }
+# glean-sym = { git = "https://github.com/mozilla/glean", tag = "v68.0.0", optional = true }
 
 [dev-dependencies]
 error-support = { path = "../support/error", features = ["testing"] }
@@ -46,4 +46,4 @@ sql-support = { path = "../support/sql" }
 
 [build-dependencies]
 uniffi = { version = "0.31", features=["build"]}
-glean-build = { git = "https://github.com/mozilla/glean", tag = "v68.0.0" }
+# glean-build = { git = "https://github.com/mozilla/glean", tag = "v68.0.0" }
diff --git a/components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt b/components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt
index 4dfc3178f..89816d107 100644
--- a/components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt
+++ b/components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt
@@ -29,12 +29,12 @@ import mozilla.appservices.places.uniffi.VisitObservation
 import mozilla.appservices.places.uniffi.VisitType
 import mozilla.appservices.places.uniffi.placesApiNew
 import mozilla.appservices.sync15.SyncTelemetryPing
-import mozilla.telemetry.glean.private.CounterMetricType
-import mozilla.telemetry.glean.private.LabeledMetricType
+// import mozilla.telemetry.glean.private.CounterMetricType
+// import mozilla.telemetry.glean.private.LabeledMetricType
 import java.lang.ref.WeakReference
 import mozilla.appservices.places.uniffi.PlacesApi as UniffiPlacesApi
 import mozilla.appservices.places.uniffi.PlacesConnection as UniffiPlacesConnection
-import org.mozilla.appservices.places.GleanMetrics.PlacesManager as PlacesManagerMetrics
+// import org.mozilla.appservices.places.GleanMetrics.PlacesManager as PlacesManagerMetrics
 
 typealias Url = String
 typealias Guid = String
@@ -312,7 +312,7 @@ class PlacesWriterConnection internal constructor(conn: UniffiPlacesConnection,
 
     @Suppress("MagicNumber")
     override fun runMaintenance(dbSizeLimit: UInt) {
-        val pruneMetrics = PlacesManagerMetrics.runMaintenanceTime.measure {
+/*        val pruneMetrics = PlacesManagerMetrics.runMaintenanceTime.measure {
             val pruneMetrics = PlacesManagerMetrics.runMaintenancePruneTime.measure {
                 this.conn.runMaintenancePrune(dbSizeLimit, 12U)
             }
@@ -330,7 +330,7 @@ class PlacesWriterConnection internal constructor(conn: UniffiPlacesConnection,
             }
             pruneMetrics
         }
-        PlacesManagerMetrics.dbSizeAfterMaintenance.accumulateSamples(listOf(pruneMetrics.dbSizeAfter.toLong() / 1024))
+        PlacesManagerMetrics.dbSizeAfterMaintenance.accumulateSamples(listOf(pruneMetrics.dbSizeAfter.toLong() / 1024)) */
     }
 
     override fun deleteEverything() {
@@ -872,13 +872,13 @@ data class HistoryMetadataKey(
  * This is a convenience wrapper to measure the two in one shot.
  */
 class PlacesManagerCounterMetrics(
-    val count: CounterMetricType,
-    val errCount: LabeledMetricType<CounterMetricType>,
+//    val count: CounterMetricType,
+//    val errCount: LabeledMetricType<CounterMetricType>,
 ) {
-    @Suppress("ComplexMethod", "TooGenericExceptionCaught")
+//    @Suppress("ComplexMethod", "TooGenericExceptionCaught")
     inline fun <U> measure(callback: () -> U): U {
-        count.add()
-        try {
+        return callback()
+/*        try {
             return callback()
         } catch (e: Exception) {
             when (e) {
@@ -905,6 +905,6 @@ class PlacesManagerCounterMetrics(
                 }
             }
             throw e
-        }
+        } */
     }
 }
-- 
