From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 05:15:52 +0000
Subject: [PATCH] Fenix - IronFox settings search

Allows users to search for IronFox-specific UI settings.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/DefaultFenixSettingsIndexer.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/DefaultFenixSettingsIndexer.kt
index 6bd919d8b664..6474f120ea9b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/DefaultFenixSettingsIndexer.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/DefaultFenixSettingsIndexer.kt
@@ -354,9 +354,8 @@ class DefaultFenixSettingsIndexer(
             PreferenceFileInformation.TabsPreferences,
             PreferenceFileInformation.TrackingProtectionPreferences,
             PreferenceFileInformation.SaveLoginsPreferences,
-            PreferenceFileInformation.DataChoicesPreferences,
-            PreferenceFileInformation.AIControlsPreferences,
             PreferenceFileInformation.FirefoxLabsPreferences,
+            PreferenceFileInformation.IronFoxPreferences,
         )
 
         /**
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/PreferenceFileInformation.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/PreferenceFileInformation.kt
index ea598d0979d4..fd90ee81b03f 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/PreferenceFileInformation.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/settingssearch/PreferenceFileInformation.kt
@@ -162,4 +162,11 @@ sealed class PreferenceFileInformation(
         categoryHeaderResourceId = R.string.preferences_category_privacy_security,
         fragmentId = R.id.trackingProtectionFragment,
     )
+
+    object IronFoxPreferences : PreferenceFileInformation(
+        xmlResourceId = R.xml.ironfox_preferences,
+        topBreadcrumbResourceId = R.string.if_preferences,
+        categoryHeaderResourceId = R.string.if_title,
+        fragmentId = R.id.ironFoxSettingsFragment,
+    )
 }
-- 
