From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 05:20:13 +0000
Subject: [PATCH] Fenix - IronFox UI

Adds minor tweaks to the browser's user interface.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
index cdfbd405ad45..7958587dec5d 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MainMenu.kt
@@ -150,6 +150,8 @@ fun MainMenu(
     webExtensionMenuCount: Int,
     showIPProtection: Boolean,
     ipProtectionMenuState: IPProtectionMenuState,
+    isPasswordManagerEnabled: Boolean,
+    isSyncActive: Boolean,
     onMoreMenuClick: () -> Unit,
     onCustomizeReaderViewMenuClick: () -> Unit,
     onMozillaAccountButtonClick: () -> Unit,
@@ -311,6 +313,7 @@ fun MainMenu(
 
         LibraryMenuGroup(
             isDownloadHighlighted = isDownloadHighlighted,
+            isPasswordManagerEnabled = isPasswordManagerEnabled,
             onBookmarksMenuClick = onBookmarksMenuClick,
             onHistoryMenuClick = onHistoryMenuClick,
             onDownloadsMenuClick = onDownloadsMenuClick,
@@ -318,11 +321,13 @@ fun MainMenu(
         )
 
         MenuGroup {
-            MozillaAccountMenuItem(
-                account = account,
-                accountState = accountState,
-                onClick = onMozillaAccountButtonClick,
-            )
+            if (isSyncActive) {
+                MozillaAccountMenuItem(
+                    account = account,
+                    accountState = accountState,
+                    onClick = onMozillaAccountButtonClick,
+                )
+            }
 
             if (accessPoint == MenuAccessPoint.Home) {
                 MenuItem(
@@ -508,6 +513,7 @@ private fun MoreMenuButtonGroup(
 @Suppress("LongMethod")
 private fun LibraryMenuGroup(
     isDownloadHighlighted: Boolean = false,
+    isPasswordManagerEnabled: Boolean = false,
     onBookmarksMenuClick: () -> Unit,
     onHistoryMenuClick: () -> Unit,
     onDownloadsMenuClick: () -> Unit,
@@ -578,16 +584,18 @@ private fun LibraryMenuGroup(
 
         Spacer(Modifier.width(spacerWidth))
 
-        LibraryMenuItem(
-            modifier = Modifier
-                .weight(1f)
-                .fillMaxHeight(),
-            iconRes = iconsR.drawable.mozac_ic_login_24,
-            labelRes = R.string.browser_menu_passwords,
-            shape = rightShape,
-            index = 3,
-            onClick = onPasswordsMenuClick,
-        )
+        if (isPasswordManagerEnabled) {
+            LibraryMenuItem(
+                modifier = Modifier
+                    .weight(1f)
+                    .fillMaxHeight(),
+                iconRes = iconsR.drawable.mozac_ic_login_24,
+                labelRes = R.string.browser_menu_passwords,
+                shape = rightShape,
+                index = 3,
+                onClick = onPasswordsMenuClick,
+            )
+        }
     }
 }
 
@@ -755,6 +763,8 @@ private fun MenuDialogPreview(
                 webExtensionMenuCount = 1,
                 showIPProtection = true,
                 ipProtectionMenuState = IPProtectionMenuState(),
+                isPasswordManagerEnabled = false,
+                isSyncActive = false,
                 onMoreMenuClick = {},
                 onCustomizeReaderViewMenuClick = {},
                 onMozillaAccountButtonClick = {},
@@ -821,6 +831,8 @@ private fun MenuDialogPrivatePreview(
                 showBanner = true,
                 isDownloadHighlighted = true,
                 webExtensionMenuCount = 0,
+                isPasswordManagerEnabled = false,
+                isSyncActive = false,
                 onMoreMenuClick = {},
                 onCustomizeReaderViewMenuClick = {},
                 onMozillaAccountButtonClick = {},
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
index ed56bde6116b..da3cf0eb0c49 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
@@ -39,6 +39,7 @@ internal fun MoreSettingsSubmenu(
     showShortcuts: Boolean,
     showSaveToCollection: Boolean,
     isAndroidAutomotiveAvailable: Boolean,
+    isPrintDisabled: Boolean,
     summarizationMenuState: SummarizationMenuState,
     isPrivate: Boolean,
     onWebCompatReporterClick: () -> Unit,
@@ -97,6 +98,7 @@ internal fun MoreSettingsSubmenu(
         )
         PrintMenuItem(
             isAndroidAutomotiveAvailable = isAndroidAutomotiveAvailable,
+            isPrintDisabled = isPrintDisabled,
             onPrintMenuClick = onPrintMenuClick,
         )
     }
@@ -231,9 +233,10 @@ private fun SaveAsPdfMenuItem(
 @Composable
 private fun PrintMenuItem(
     isAndroidAutomotiveAvailable: Boolean,
+    isPrintDisabled: Boolean,
     onPrintMenuClick: () -> Unit,
 ) {
-    if (!isAndroidAutomotiveAvailable) {
+    if (!isPrintDisabled && !isAndroidAutomotiveAvailable) {
         MenuItem(
             label = stringResource(id = R.string.browser_menu_print_2),
             beforeIconPainter = painterResource(id = iconsR.drawable.mozac_ic_print_24),
@@ -325,6 +328,7 @@ private fun MoreSettingsSubmenuPreview(
                     showShortcuts = true,
                     showSaveToCollection = true,
                     isAndroidAutomotiveAvailable = false,
+                    isPrintDisabled = false,
                     summarizationMenuState = SummarizationMenuState.Default.copy(
                         visible = true,
                         highlighted = true,
@@ -378,6 +382,7 @@ private fun MoreSettingsSubmenuDisabledOpenPreview(
                     showShortcuts = true,
                     showSaveToCollection = true,
                     isAndroidAutomotiveAvailable = false,
+                    isPrintDisabled = false,
                     summarizationMenuState = SummarizationMenuState.Default,
                     isPrivate = false,
                     onWebCompatReporterClick = {},
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
index 5634befbf0d0..53a4aa8166aa 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/MenuDialogFragment.kt
@@ -528,7 +528,7 @@ class MenuDialogFragment : BottomSheetDialogFragment() {
                                     accessPoint = args.accesspoint,
                                     account = account,
                                     accountState = accountState,
-                                    showQuitMenu = settings.shouldDeleteBrowsingDataOnQuit,
+                                    showQuitMenu = true,
                                     isBottomToolbar = settings.shouldUseBottomToolbar,
                                     isExpandedToolbarEnabled = settings.shouldUseExpandedToolbar,
                                     isSiteLoading = isSiteLoading,
@@ -548,6 +548,8 @@ class MenuDialogFragment : BottomSheetDialogFragment() {
                                     showBanner = shouldShowMenuBanner && !defaultBrowser,
                                     isDownloadHighlighted = isDownloadHighlighted,
                                     webExtensionMenuCount = webExtensionsCount,
+                                    isPasswordManagerEnabled = settings.shouldPromptToSaveLogins,
+                                    isSyncActive = settings.signedInFxaAccount,
                                     isAllWebExtensionsDisabled = isAllWebExtensionsDisabled,
                                     showIPProtection = components.ipProtection.store.state.isEligible,
                                     ipProtectionMenuState = ipProtectionMenuState,
@@ -658,9 +660,10 @@ class MenuDialogFragment : BottomSheetDialogFragment() {
                                             externalAppName = appLinksRedirect?.appName ?: "",
                                             isOpenInAppMenuHighlighted = isOpenInAppMenuHighlighted,
                                             translationInfo = translationInfo,
-                                            showShortcuts = settings.showTopSitesFeature,
+                                            showShortcuts = settings.showTopSitesFeature && !settings.ironfox.alwaysUsePrivateBrowsing,
                                             showSaveToCollection = settings.collections,
                                             isAndroidAutomotiveAvailable = context.isAndroidAutomotiveAvailable(),
+                                            isPrintDisabled = !settings.ironfox.printEnabled,
                                             summarizationMenuState = summarizationMenuState,
                                             onWebCompatReporterClick = {
                                                 menuStore.dispatch(MenuAction.Navigate.WebCompatReporter)
-- 
