From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Fenix - Configure GeckoProvider

Configures Firefox's implementation of GeckoView (GeckoProvider).

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
index 0bca79f30c33..30c1f9b40278 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/gecko/GeckoProvider.kt
@@ -7,14 +7,14 @@ package org.mozilla.fenix.gecko
 import android.content.Context
 import androidx.annotation.VisibleForTesting
 import mozilla.components.browser.engine.gecko.autofill.GeckoAutocompleteStorageDelegate
-import mozilla.components.browser.engine.gecko.crash.GeckoCrashPullDelegate
+// import mozilla.components.browser.engine.gecko.crash.GeckoCrashPullDelegate
 import mozilla.components.browser.engine.gecko.ext.toContentBlockingSetting
 import mozilla.components.concept.engine.EngineSession.TrackingProtectionPolicy
 import mozilla.components.concept.storage.CreditCardsAddressesStorage
 import mozilla.components.concept.storage.LoginsStorage
-import mozilla.components.experiment.NimbusExperimentDelegate
+/* import mozilla.components.experiment.NimbusExperimentDelegate
 import mozilla.components.lib.crash.handler.CrashHandlerService
-import mozilla.components.lib.crash.store.CrashAction
+import mozilla.components.lib.crash.store.CrashAction */
 import mozilla.components.service.sync.autofill.GeckoCreditCardsAddressesStorageDelegate
 import mozilla.components.service.sync.logins.GeckoLoginStorageDelegate
 import org.mozilla.fenix.Config
@@ -72,13 +72,13 @@ object GeckoProvider {
             ),
         )
 
-        geckoRuntime.crashPullDelegate = GeckoCrashPullDelegate(
+/*        geckoRuntime.crashPullDelegate = GeckoCrashPullDelegate(
             dispatcher = { crashIDs ->
                 context.components.appStore.dispatch(
                     AppAction.CrashActionWrapper(CrashAction.CheckDeferred(crashIDs.toList())),
                 )
             },
-        )
+        ) */
 
         return geckoRuntime
     }
@@ -89,8 +89,8 @@ object GeckoProvider {
         policy: TrackingProtectionPolicy,
     ): GeckoRuntimeSettings {
         val builder = GeckoRuntimeSettings.Builder()
-            .crashHandler(CrashHandlerService::class.java)
-            .experimentDelegate(NimbusExperimentDelegate())
+//            .crashHandler(CrashHandlerService::class.java)
+//            .experimentDelegate(NimbusExperimentDelegate())
             .contentBlocking(
                 policy.toContentBlockingSetting(
                     cookieBannerHandlingMode = context.settings().getCookieBannerHandling(),
-- 
