From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Fenix - Remove Web Compatibility Reporter

Removes the Web Compatibility Reporter.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/.buildconfig.yml b/mobile/android/fenix/.buildconfig.yml
index e23ec109a522..dfaba7eaf03a 100644
--- a/mobile/android/fenix/.buildconfig.yml
+++ b/mobile/android/fenix/.buildconfig.yml
@@ -71,7 +71,7 @@ projects:
     - components:feature-top-sites
     - components:feature-webauthn
     - components:feature-webcompat
-    - components:feature-webcompat-reporter
+#    - components:feature-webcompat-reporter
     - components:feature-webnotifications
     - components:lib-accelerometer-sensormanager
     - components:lib-ai-controls
diff --git a/mobile/android/fenix/app/build.gradle b/mobile/android/fenix/app/build.gradle
index ad28bfb8c8ff..a97427c7925c 100644
--- a/mobile/android/fenix/app/build.gradle
+++ b/mobile/android/fenix/app/build.gradle
@@ -590,7 +590,7 @@ dependencies {
     implementation project(':components:feature-top-sites')
     implementation project(':components:feature-webauthn')
     implementation project(':components:feature-webcompat')
-    implementation project(':components:feature-webcompat-reporter')
+//    implementation project(':components:feature-webcompat-reporter')
     implementation project(':components:feature-webnotifications')
     implementation project(':components:service-digitalassetlinks')
     implementation project(':components:service-firefox-accounts')
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
index ed56bde6116b..cfb05bdd8c01 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/components/menu/compose/MoreSettingsSubmenu.kt
@@ -68,10 +68,10 @@ internal fun MoreSettingsSubmenu(
             isPrivate = isPrivate,
             onMoveToNonPrivateTabMenuClick = onMoveToNonPrivateTabMenuClick,
         )
-        WebCompatReporterMenuItem(
+/*        WebCompatReporterMenuItem(
             isWebCompatEnabled = isWebCompatEnabled,
             onWebCompatReporterClick = onWebCompatReporterClick,
-        )
+        ) */
         ShortcutsSection(
             showShortcuts = showShortcuts,
             isPinned = isPinned,
-- 
