From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Fenix - Configure DNS over HTTPS providers

Adds custom DoH providers.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/doh/DohSettingsProvider.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/doh/DohSettingsProvider.kt
index 028be832d6d7..4cbee0999688 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/doh/DohSettingsProvider.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/doh/DohSettingsProvider.kt
@@ -37,17 +37,42 @@ internal class DefaultDohSettingsProvider(
 
     private val cloudflare = Provider.BuiltIn(
         url = cloudflareUri,
-        name = "Cloudflare",
-        default = dohDefaultProviderUrl.isBlank() || dohDefaultProviderUrl == cloudflareUri,
+        name = "Cloudflare 🇺🇸",
+        default = dohDefaultProviderUrl == cloudflareUri,
     )
-    private val nextDns = Provider.BuiltIn(
-        url = nextDnsUri,
-        name = "NextDNS",
-        default = dohDefaultProviderUrl == nextDnsUri,
+    private val mullvadBase = Provider.BuiltIn(
+        url = mullvadBaseUri,
+        name = "Mullvad (Base) 🇸🇪",
+        default = dohDefaultProviderUrl.isNullOrBlank() || dohDefaultProviderUrl == mullvadBaseUri,
+    )
+    private val cloudflareMalware = Provider.BuiltIn(
+        url = cloudflareMalwareUri,
+        name = "Cloudflare (Malware Protection) 🇺🇸",
+        default = dohDefaultProviderUrl == cloudflareMalwareUri,
+    )
+    private val dns4EuAdBlocking = Provider.BuiltIn(
+        url = dns4EuAdBlockingUri,
+        name = "DNS4EU (Ad Blocking) 🇨🇿",
+        default = dohDefaultProviderUrl == dns4EuAdBlockingUri,
+    )
+    private val dns4EuProtective = Provider.BuiltIn(
+        url = dns4EuProtectiveUri,
+        name = "DNS4EU (Protective) 🇨🇿",
+        default = dohDefaultProviderUrl == dns4EuProtectiveUri,
+    )
+    private val dns4EuUnfiltered = Provider.BuiltIn(
+        url = dns4EuUnfilteredUri,
+        name = "DNS4EU (Unfiltered) 🇨🇿",
+        default = dohDefaultProviderUrl == dns4EuUnfilteredUri,
+    )
+    private val mullvadUnfiltered = Provider.BuiltIn(
+        url = mullvadUnfilteredUri,
+        name = "Mullvad (Unfiltered) 🇸🇪",
+        default = dohDefaultProviderUrl == mullvadUnfilteredUri,
     )
     private val providerUrl = settings.dohProviderUrl
     private val custom = Provider.Custom(
-        url = if (providerUrl != cloudflareUri && providerUrl != nextDnsUri) {
+        url = if (providerUrl != cloudflareUri && providerUrl != mullvadBaseUri && providerUrl != cloudflareMalwareUri && providerUrl != dns4EuAdBlockingUri && providerUrl != dns4EuProtectiveUri && providerUrl != dns4EuUnfilteredUri && providerUrl != mullvadUnfilteredUri) {
             providerUrl
         } else {
             ""
@@ -55,8 +80,13 @@ internal class DefaultDohSettingsProvider(
     )
 
     override fun getDefaultProviders(): List<Provider> = listOf(
+        mullvadBase,
+        cloudflareMalware,
         cloudflare,
-        nextDns,
+        dns4EuAdBlocking,
+        dns4EuProtective,
+        dns4EuUnfiltered,
+        mullvadUnfiltered,
         custom,
     )
 
@@ -78,7 +108,12 @@ internal class DefaultDohSettingsProvider(
             else -> {
                 when (settings.dohProviderUrl) {
                     cloudflareUri -> cloudflare
-                    nextDnsUri -> nextDns
+                    mullvadBaseUri -> mullvadBase
+                    cloudflareMalwareUri -> cloudflareMalware
+                    dns4EuAdBlockingUri -> dns4EuAdBlocking
+                    dns4EuProtectiveUri -> dns4EuProtective
+                    dns4EuUnfilteredUri -> dns4EuUnfiltered
+                    mullvadUnfilteredUri -> mullvadUnfiltered
                     "" -> getDefaultProviders().first()
                     else -> custom
                 }
@@ -121,6 +156,11 @@ internal class DefaultDohSettingsProvider(
         val cloudflareUri = "https://mozilla.cloudflare-dns.com/dns-query"
 
         @VisibleForTesting
-        val nextDnsUri = "https://firefox.dns.nextdns.io/"
+        val mullvadBaseUri = "https://base.dns.mullvad.net/dns-query"
+        val cloudflareMalwareUri = "https://security.cloudflare-dns.com/dns-query"
+        val dns4EuAdBlockingUri = "https://noads.joindns4.eu/dns-query"
+        val dns4EuProtectiveUri = "https://protective.joindns4.eu/dns-query"
+        val dns4EuUnfilteredUri = "https://unfiltered.joindns4.eu/dns-query"
+        val mullvadUnfilteredUri = "https://dns.mullvad.net/dns-query"
     }
 }
-- 
