From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sat, 14 Feb 2026 04:05:55 +0000
Subject: [PATCH] Fenix - Disable contextual feature recommendations

Disables contextual feature recommendations (CFRs) and other promotional messaging.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 8113d3cfa7bc..42939460d935 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -391,7 +391,7 @@ class Settings(
     )
 
     val canShowCfr: Boolean
-        get() = (currentTimeMillis() - lastCfrShownTimeInMillis) > THREE_DAYS_MS
+        get() = false
 
     val cfrPopupsEnabled by booleanPreference(
         appContext.getPreferenceKey(R.string.pref_cfr_popups_enabled),
-- 
