From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] GeckoView - Liberate

Replaces the Google Play FIDO library with microG.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/geckoview/build.gradle b/mobile/android/geckoview/build.gradle
index e12d8b201d..48fbffef3a 100644
--- a/mobile/android/geckoview/build.gradle
+++ b/mobile/android/geckoview/build.gradle
@@ -192,7 +192,7 @@ dependencies {
     implementation libs.androidx.core
     implementation libs.androidx.lifecycle.common
     implementation libs.androidx.lifecycle.process
-    implementation libs.play.services.fido
+    implementation "org.microg.gms:play-services-fido:v0.0.0.252432"
     implementation "org.yaml:snakeyaml:2.2"
 
     if (mozconfig.substs.MOZ_ANDROID_HLS_SUPPORT) {
-- 
