From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Expose hidden settings at about:support

Enables the display of advanced options at `about:support`.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/toolkit/content/aboutSupport.js b/toolkit/content/aboutSupport.js
index fb4fbac6ac59..0d300fd2c8b7 100644
--- a/toolkit/content/aboutSupport.js
+++ b/toolkit/content/aboutSupport.js
@@ -551,7 +551,7 @@ var snapshotFormatters = {
 
     if (gpuProcessPid != -1) {
       let gpuProcessKillButton = null;
-      if (AppConstants.NIGHTLY_BUILD || AppConstants.MOZ_DEV_EDITION) {
+      if (true) {
         gpuProcessKillButton = $.new("button");
 
         gpuProcessKillButton.addEventListener("click", function () {
@@ -571,7 +571,7 @@ var snapshotFormatters = {
     }
 
     if (
-      (AppConstants.NIGHTLY_BUILD || AppConstants.MOZ_DEV_EDITION) &&
+      (true) &&
       AppConstants.platform != "macosx"
     ) {
       let gpuDeviceResetButton = $.new("button");
diff --git a/toolkit/content/aboutSupport.xhtml b/toolkit/content/aboutSupport.xhtml
index 021e6a326703..e324c3ce1e49 100644
--- a/toolkit/content/aboutSupport.xhtml
+++ b/toolkit/content/aboutSupport.xhtml
@@ -43,8 +43,8 @@
         </div>
       </div>
 
-#ifndef ANDROID
       <div class="action-box">
+#ifndef ANDROID
         <div id="reset-box" hidden="hidden">
           <h3 data-l10n-id="refresh-profile"/>
           <button id="reset-box-button" data-l10n-id="refresh-profile-button"/>
@@ -53,12 +53,12 @@
           <h3 data-l10n-id="troubleshoot-mode-title"/>
           <button id="restart-in-safe-mode-button" data-l10n-id="restart-in-troubleshoot-mode-label"/>
         </div>
+#endif
         <div id="clear-startup-cache-box">
           <h3 data-l10n-id="clear-startup-cache-title"/>
           <button id="clear-startup-cache-button" data-l10n-id="clear-startup-cache-label"/>
         </div>
       </div>
-#endif
     </div>
     <div id="contents">
 
-- 
