From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Enable about:inference

Enables the `about:inference` page.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/docshell/base/nsAboutRedirector.cpp b/docshell/base/nsAboutRedirector.cpp
index 76f707e5f1e8e..fdfcd2ab2d0a0 100644
--- a/docshell/base/nsAboutRedirector.cpp
+++ b/docshell/base/nsAboutRedirector.cpp
@@ -134,7 +134,7 @@ static const RedirEntry kRedirMap[] = {
      nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT |
          nsIAboutModule::URI_CAN_LOAD_IN_CHILD | nsIAboutModule::ALLOW_SCRIPT |
          nsIAboutModule::HIDE_FROM_ABOUTABOUT},
-#if defined(NIGHTLY_BUILD)
+#if true
     {"inference", "chrome://global/content/aboutInference.html",
      nsIAboutModule::ALLOW_SCRIPT | nsIAboutModule::IS_SECURE_CHROME_UI},
 #endif
diff --git a/docshell/build/components.conf b/docshell/build/components.conf
index 76a6380f849c2..b7cb7d16db014 100644
--- a/docshell/build/components.conf
+++ b/docshell/build/components.conf
@@ -51,7 +51,7 @@ if not defined('MOZ_GLEAN_ANDROID'):
     about_pages.append('glean')
 if buildconfig.substs['MOZ_WIDGET_TOOLKIT'] != 'android' and buildconfig.substs['MOZ_WIDGET_TOOLKIT'] != 'windows':
     about_pages.append('webauthn')
-if defined('NIGHTLY_BUILD'):
+if True:
     about_pages.append('inference')
 
 Headers = ['/docshell/build/nsDocShellModule.h']
diff --git a/toolkit/components/moz.build b/toolkit/components/moz.build
index fc14fc3bd7d14..87af54c162c51 100644
--- a/toolkit/components/moz.build
+++ b/toolkit/components/moz.build
@@ -150,7 +150,7 @@ if CONFIG["MOZ_BACKGROUNDTASKS"]:
 if CONFIG["MOZ_UNIFFI_FIXTURES"]:
     DIRS += ["uniffi-bindgen-gecko-js/tests"]
 
-if CONFIG["NIGHTLY_BUILD"]:
+if True:
     DIRS += ["aboutinference"]
 
 UNIFIED_SOURCES += [
diff --git a/toolkit/locales/jar.mn b/toolkit/locales/jar.mn
index 480dbd7db7981..522e98f0928dd 100644
--- a/toolkit/locales/jar.mn
+++ b/toolkit/locales/jar.mn
@@ -12,7 +12,7 @@
 #ifdef MOZ_LAYOUT_DEBUGGER
   layoutdebug/layoutdebug.ftl                      (../../layout/tools/layout-debug/ui/content/layoutdebug.ftl)
 #endif
-#if defined(NIGHTLY_BUILD)
+#if !defined(DER_NOOP)
   preview/aboutInference.ftl                       (../components/aboutinference/content/aboutInference.ftl)
 #endif
 
-- 
