From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] GeckoView - Disable speculative connections

Prevents GeckoView from generating speculative/predictive connections.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/mobile/android/geckoview/src/main/java/org/mozilla/gecko/GeckoThread.java b/mobile/android/geckoview/src/main/java/org/mozilla/gecko/GeckoThread.java
index a80a784ff1659..d9057f8fa57f0 100644
--- a/mobile/android/geckoview/src/main/java/org/mozilla/gecko/GeckoThread.java
+++ b/mobile/android/geckoview/src/main/java/org/mozilla/gecko/GeckoThread.java
@@ -751,7 +751,7 @@ public class GeckoThread extends Thread {
     // bother checking here if Gecko is actually loaded or not.
     // Speculative connection depends on proxy settings,
     // so the earliest it can happen is after profile is ready.
-    queueNativeCallUntil(State.PROFILE_READY, GeckoThread.class, "speculativeConnectNative", uri);
+//    queueNativeCallUntil(State.PROFILE_READY, GeckoThread.class, "speculativeConnectNative", uri);
   }
 
   @UiThread
-- 
