From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Prevent fingerprinting via EME

Prevents fingerprinting via the status of EME.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/dom/media/eme/MediaKeySystemAccessManager.cpp b/dom/media/eme/MediaKeySystemAccessManager.cpp
index acdc1030c06fa..c686a80691077 100644
--- a/dom/media/eme/MediaKeySystemAccessManager.cpp
+++ b/dom/media/eme/MediaKeySystemAccessManager.cpp
@@ -421,8 +421,8 @@ void MediaKeySystemAccessManager::RequestMediaKeySystemAccess(
     return;
   }
 
-  if (!StaticPrefs::media_eme_enabled() &&
-      !IsClearkeyKeySystem(aRequest->mKeySystem)) {
+  if (!StaticPrefs::media_eme_enabled()) {
+//      !IsClearkeyKeySystem(aRequest->mKeySystem)) {
     // EME disabled by user, send notification to chrome so UI can inform user.
     // Clearkey is allowed even when EME is disabled because we want the pref
     // "media.eme.enabled" only taking effect on proprietary DRMs.
@@ -431,7 +431,7 @@ void MediaKeySystemAccessManager::RequestMediaKeySystemAccess(
       MediaKeySystemAccess::NotifyObservers(mWindow, aRequest->mKeySystem,
                                             MediaKeySystemStatus::Api_disabled);
     }
-    aRequest->RejectPromiseWithNotSupportedError("EME has been preffed off"_ns);
+    aRequest->RejectPromiseWithNotSupportedError("The application embedding this user agent has blocked MediaKeySystemAccess"_ns);
     aRequest->mDiagnostics.StoreMediaKeySystemAccess(
         mWindow->GetExtantDoc(), aRequest->mKeySystem, false, __func__);
     return;
-- 
